src/room/GridLayoutMenu.tsx

```diff
@@ -46,7 +46,12 @@ export function GridLayoutMenu({ layout, setLayout }: Props) {
         </Button>
       </TooltipTrigger>
       {(props: JSX.IntrinsicAttributes) => (
-        <Menu {...props} label={t("Grid layout menu")} onAction={setLayout}>
+        <Menu
+          {...props}
+          label={t("Grid layout menu")}
+          onAction={(key) => setLayout(key.toString() as Layout)}
+          onClose={() => {}}
```

Comment:
Good catch, fixed!

OOC: Do you by chance know if there is a lint for that or something similar so that such things could be caught in advance? I'm always forgetting about these sorts of things (not a frontend/React developer 🙂).

Reply:
I don't think so, sadly… but you're very right that it's easy to forget these things. Usually it's not a big deal to leave things unmemoized, but Meta is also experimenting with an auto-memoizing compiler that could improve the developer experience here.